android/vcmi-app/src/main/java/eu/vcmi/vcmi/util/FileUtil.java

```diff
@@ -104,4 +113,50 @@ private static void copyStream(InputStream source, OutputStream target) throws I
             target.write(buffer, 0, read);
         }
     }
+
+    @SuppressWarnings(Const.JNI_METHOD_SUPPRESS)
+    private static void copyFileFromUri(String sourceFileUri, String destinationFile)
+    {
+        try
+        {
+            final Context ctx = SDL.getContext();
```

Comment:
pass context from C++ as main activity, Qt provides a method to get it. Treat SDL as non-existant in the launcher.